### PR TITLE
Add private route for restore workbook

### PR DESCRIPTION
**Pull request by Sergey-weber**

### CODE REVIEW

File: src/services/new/workbook/restore-workbook.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        validateArgs(args);
+    }
+
+    const {tenantId} = ctx.get('info');
+
+    const targetTrx = getReplica(trx);
+
+    const model: Optional<WorkbookModel> = await WorkbookModel.query(targetTrx)
+        .select()
+        .skipUndefined()
```
Comment (imsitnikov) [inline src/services/new/workbook/restore-workbook.ts:46]: Is it needed here and below?

File: src/services/new/workbook/restore-workbook.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        .where({
+            [WorkbookModelColumn.TenantId]: tenantId,
+            [WorkbookModelColumn.WorkbookId]: workbookId,
+        })
+        .andWhereNot(WorkbookModelColumn.DeletedAt, null)
```
Comment (imsitnikov) [inline src/services/new/workbook/restore-workbook.ts:51]: Let's check this after the query, like:

```typescript
if (!model) {
    throw new AppError(US_ERRORS.WORKBOOK_NOT_EXISTS, {
        code: US_ERRORS.WORKBOOK_NOT_EXISTS,
    });
}

if (model.deletedAt !== null) {
    throw new AppError(US_ERRORS.WORKBOOK_IS_ALREADY_RESTORED, { // new error with 400 status
        code: US_ERRORS.WORKBOOK_IS_ALREADY_RESTORED,
    });
}
```

File: src/services/new/workbook/restore-workbook.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            })
+            .where({
+                [WorkbookModelColumn.WorkbookId]: model.workbookId,
+                [WorkbookModelColumn.TenantId]: tenantId,
+            })
+            .returning('*')
+            .first()
+            .timeout(WorkbookModel.DEFAULT_QUERY_TIMEOUT);
+
+        await Promise.all(
```
Comment (imsitnikov) [inline src/services/new/workbook/restore-workbook.ts:89]: We can restore all  the entries by one patch query:

```sql
UPDATE entries SET
	key = regexp_replace(key, '__trash/', ''),
	display_key = regexp_replace(display_key, '__trash/', ''),
	inner_meta = inner_meta - 'oldKey' - 'oldDisplayKey',
	is_deleted = FALSE,
	deleted_at = NULL,
	updated_at = NOW()
WHERE
	tenant_id = '{TENANT_ID}' AND workbook_id = {WORKBOOK_ID} AND deleted_at > '{WORKBOOK_DELETED_AT}';
```

And getting all the entries above will be unnecessary.

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/services/new/workbook/restore-workbook.ts (hunk 1 above)
Comment (Sergey-weber) [reply src/services/new/workbook/restore-workbook.ts]: unnessesary, fixed it, was error when don't have undefined

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**